Source/WebKit/UIProcess/API/APINavigation.cpp

```diff
@@ -113,6 +114,12 @@ void Navigation::appendRedirectionURL(const WTF::URL& url)
         m_redirectChain.append(url);
 }
 
+bool Navigation::currentRequestIsCrossOriginRedirect() const
```

Comment:
Based on the implementation, it sounds like you mean cross-site, not cross-origin.